haskell/src/Lorentz/Contracts/Stablecoin.hs

```diff
@@ -507,68 +471,194 @@ metadataMap mdata = BigMap $ Map.fromList $
-        , mkError [mt|NOT_MASTER_MINTER|]          [mt|Operation can only be performed by the contract's master minter|]
-        , mkError [mt|NOT_MINTER|]                 [mt|Operation can only be performed by registered minters|]
-        , mkError [mt|CONTRACT_PAUSED|]            [mt|Operation cannot be performed while the contract is paused|]
-        , mkError [mt|CONTRACT_NOT_PAUSED|]        [mt|Operation cannot be performed while the contract is not paused|]
-        , mkError [mt|NOT_TOKEN_OWNER|]            [mt|You cannot configure another user's operators|]
-        , mkError [mt|CURRENT_ALLOWANCE_REQUIRED|] [mt|The given address is already a minter, you must specify its current minting allowance|]
-        , mkError [mt|ALLOWANCE_MISMATCH|]         [mt|The given current minting allowance does not match the minter's actual current minting allowance|]
-        , mkError [mt|ADDR_NOT_MINTER|]            [mt|This address is not a registered minter|]
-        , mkError [mt|ALLOWANCE_EXCEEDED|]         [mt|The amount of tokens to be minted exceeds your current minting allowance|]
-        , mkError [mt|BAD_TRANSFERLIST|]           [mt|The given address is a not a smart contract complying with the transferlist interface|]
-        , mkError [mt|MINTER_LIMIT_REACHED|]       [mt|Cannot add new minter because the number of minters is already at the limit|]
-        , mkError [mt|MISSIGNED|]                  [mt|This permit's signature is invalid|]
-        , mkError [mt|EXPIRED_PERMIT|]             [mt|A permit was found, but it has already expired|]
-        , mkError [mt|NOT_PERMIT_ISSUER|]          [mt|You're not the issuer of the given permit|]
-        , mkError [mt|DUP_PERMIT|]                 [mt|The given permit already exists|]
-        , mkError [mt|EXPIRY_TOO_BIG|]             [mt|The `set_expiry` entrypoint was called with an expiry value that is too big|]
-        ]
-    , mViews =
-        [ getDefaultExpiryView
-        , getCounterView
-        ]
-    }
+-- | Make the TZIP-16 metadata. We accept a Maybe FA2.Token metadata
+-- as argument here so that we can use this function to create the metadata of the
+-- FA1.2 Variant as well.
+metadataJSON :: Maybe FA2.TokenMetadata -> Metadata (ToT Storage)
+metadataJSON mtmd  =
+  TZ.name  "stablecoin" <>
+  TZ.version (toText $ showVersion version) <>
+  TZ.license (License { lName = "MIT", lDetails = Nothing }) <>
+  TZ.authors
+      [ TZ.author "Serokell" "https://serokell.io/"
+      , TZ.author "TQ Tezos" "https://tqtezos.com/"
+      ] <>
+  TZ.homepage "https://github.com/tqtezos/stablecoin/" <>
+  TZ.source Source
+     { sLocation = "https://github.com/tqtezos/stablecoin/tree/v" <> toText (showVersion version) <> "/ligo/stablecoin"
+     , sTools = [ "ligo " ] -- TODO: add ligo version
+     } <>
+  TZ.interfaces [ TZ.Interface "TZIP-12", TZ.Interface "TZIP-17" ] <>
```

Comment:
I noticed that here it's `TZ.Interface "TZIP-12"` which maybe was
ok beforehand, but it seems that as a result of the [recent update](https://gitlab.com/tzip/tzip/-/merge_requests/94) this is now more precisely defined as being `TZIP-012` (with a `0`) in the [TZIP-016 section](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-12/tzip-12.md#contract-metadata-tzip-016):
> It should contain "TZIP-012[-<version-info>]"

so I think we should update it.

In general it seems that the `0` is present in the [TZIP-16 own specification](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md) too everywhere, so I suggest we update `TZ.Interface "TZIP-17"` as well.

Reply:
Have change the interface names in metadata to have the zero prefix.